src/java/org/apache/cassandra/db/SerializationHeader.java

```diff
@@ -324,7 +324,7 @@ private static AbstractType<?> validateType(String descriptor,
             boolean dropped = table.getDroppedColumn(columnName) != null;
             if (!dropped && type.isTuple() && type.isMultiCell())
             {
-                logger.error("Error reading SSTable header {}, the type for column {} in {} is not-frozen {}, " +
+                logger.debug("Error reading SSTable header {}, the type for column {} in {} is not-frozen {}, " +
```

Comment:
If I'm not missing something, I think we can change [this exception message](https://github.com/datastax/cassandra/pull/998) to use `toSchemaString` and get rid of this check. 

That way, all the log messages are printed by the branch that used WARN, the one with a fixed type. Also, since the branch without the ERROR level is not hit by regular old sstables, we might want to keep it with ERROR or WARN.

Reply:
The link is broken, I'm not sure if I can understand your intention. However, as long as we are not throwing an exception leading to skip the sstable, I think it is ok to log it as debug. It is just an information that something go fixed. Maybe we should simply bump sstable version and get rid of those messages entirely?

Reply:
Actually, I'm convinced we should do that this way - new serialization, new interpretation --> new sstable format version. We can then conclude what we are reading and what to expect from a certain version.

Reply:
Sorry for the broken link, this is what I meant: https://github.com/adelapena/cassandra/commit/8cc0be87222c852eaf425a7cdbc5c4b2bf0b5e6e

Reply:
[This error](https://github.com/datastax/cassandra/blob/75fbf164662c14a26ee513317c0df94678cc4074/src/java/org/apache/cassandra/db/SerializationHeader.java#L342-L351) happens when we haven't found a suitable replacement type and we are just ignoring the error, so it might make sense to use WARN. Note that the new test doesn't hit that error with the suggested changes.

I agree that bumping the sstable version makes sense, given that we are introducing a new semantic, even though we are able to keep working with the old one.

Reply:
Ok, I was thinking about that more - what makes me worry about all this stuff:
- what will happen if an old node attempts to read the new sstable with a dropped columns record that specify unfrozen tuple for some data in a deleted column?

I guess it will interpret it as frozen and we end up with corruption, am I right?

Why would it attempt to read a newer sstable? well - because there is assumption we can read a newer sstable in an old node given it is a just a minor version change. So, in this case we would have to switch to the next major version, which would prevent reading the new sstables entirely by the old nodes. Switching to the next major SSTable version, means version D of BTI, which is already occupied by OSS 5.0, and is something else than what we would interpret as D. Also, it feels the problem has a similar nature as the introduction of OA format in OSS.

Does the above reasoning make any sense?

Reply:
On the other hand, I can merge this fix and let us debate on the above problem separately

Reply:
@adelapena that case you mentioned should make the sstable fail to load; otherwise, it may result in corrupted reads